### ISSUE TITLE
Pattern Match Failure when calling maybe_subscribe/2

### ISSUE THREAD

**Issue by luckyluke578** [pentacent/phoenix_live_session#6]
Hi there

Thank you for your package, Phoenix_live_session, it is a great help.

I recently upgraded from v0.1.1 to v0.1.2 and now I get the following error when calling maybe_subscribe:
 
```
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in PhoenixLiveSession.maybe_subscribe/2

```
The call site is in the mount/3 function of my live component:

```
def mount(_params, session, socket) do
    socket =
      socket
      |> maybe_subscribe(session)

    IO.inspect session

    {:ok, socket}
  end

```
If I edit phoenix_live_session.ex in my deps folder and revert the changes to maybe_subscribe as follows, then it works again:

```
def maybe_subscribe(socket, session) do
    if LiveView.connected?(socket) do
      opts = session[:__opts__]
      sid = session[:__sid__]
      pub_sub = Keyword.fetch!(opts, :pub_sub)
      channel = "live_session:#{sid}"
      PubSub.subscribe(pub_sub, channel)

      put_in(socket.private[:live_session], id: sid, opts: opts)
    else
      socket
    end
  end

```
The mount/3 call then succeeds, and produces the following output for 'session' via IO.inspect:

```
%{
  :__opts__ => [
    clean_interval: 60000,
    lifetime: 172800000,
    table: :phoenix_live_sessions,
    pub_sub: XXXX.PubSub,
    signing_salt: "XXXX"
  ],
  :__sid__ => "XXXX",
  "_csrf_token" => "XXXX"
}
```

Thanks in advance for any help you may provide. If you need any further info about the issue, please let me know..

**Comment by wmnnd** [comment]
Did you set up LiveSession in your `Endpoint` module? The error sounds like the session isn’t actually coming from the LiveSession store.

**Comment by jvantuyl** [comment]
I'm currently having the same problem.  Looks like it's getting called with a disconnected socket and an empty map for the session.  This causes the match to fail well before it ever checks to see if it's connected at all.  I think this may be happening because I never hit a page that generates the session before I try to render the live view (unauthenticated app).  Perhaps that's unusual?

Here's the error, with some redaction of sensitive info:
```
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in PhoenixLiveSession.maybe_subscribe/2
        (phoenix_live_session 0.1.2) lib/phoenix_live_session.ex:228: PhoenixLiveSession.maybe_subscribe(#Phoenix.LiveView.Socket<assigns: %{flash: %{}, live_action: :orient}, changed: %{}, endpoint: MyAppWeb.Endpoint, id: "phx-FoZyS98su9gWw5oH", parent_pid: nil, root_pid: nil, router: MyAppWeb.Router, transport_pid: nil, view: MyAppWeb.OrientLive, ...>, %{})
```

**Comment by luckyluke578** [comment]
> Did you set up LiveSession in your `Endpoint` module? The error sounds like the session isn’t actually coming from the LiveSession store.

Hi there,

Thank you so much for responding. I did indeed set the LiveSession up in endpoint.ex:

```
@session_options [
    # store: :cookie,
    store: PhoenixLiveSession,
    pub_sub: XXXX.PubSub,
    key: "_XXXX_XXXX_key",
    signing_salt: "XXXXXXXX"
  ]
```

I followed what the documentation set out, if that helps.

**Comment by wmnnd** [comment]
Fixed via #7.
